src/stellar/StellarTools.ts

```diff
@@ -81,11 +84,28 @@ export class StellarTools implements EdgeCurrencyTools {
     return { stellarKey: keypair.secret() }
   }
 
-  async importPrivateKey(privateKey: string): Promise<{ stellarKey: string }> {
-    privateKey.replace(/ /g, '')
-    stellarApi.Keypair.fromSecret(privateKey)
-    if (privateKey.length !== 56) throw new Error('Private key wrong length')
-    return await Promise.resolve({ stellarKey: privateKey })
+  async importPrivateKey(userInput: string): Promise<StellarPrivateKeys> {
+    let stellarKey
+    let stellarMnemonic
+
+    if (validateMnemonic(userInput)) {
+      const derivedSeed = ed25519.derivePath("m/44'/314159'/0'", userInput).key
+      const keypair = stellarApi.Keypair.fromRawEd25519Seed(
+        Uint8Array.from(derivedSeed) as unknown as number[]
+      )
+      stellarKey = keypair.secret().replace(/\s/g, '')
```

Comment:
Why are you editing the key returned from secret()? We use this to instantiate a KeyPair later so it needs to be the exact value